### PR TITLE
Optimize checkout

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR introduces explicit checkout steps to the pipeline to optimize the pipelines. Where possible we will eliminate checkout.

### CODE REVIEW

**Comment by JonathanGiles** [comment]
Checkout is much faster across the board now - looks good to me.

**Comment by JonathanGiles** [comment]
It does appear that all branches are being cloned - is it worth just configuring it to clone master only?

**Comment by mitchdenny** [comment]
Yep - a fair chunk of time is spent on the server building a set of pack files to transfer to the client and I was wondering whether we could optimize by having a greater fetch depth to trick GitHub into reusing pack files but its probably diminishing returns - and the gh-pages thing will have a bigger impact anyway.

**Comment by mitchdenny** [comment]
> It does appear that all branches are being cloned - is it worth just configuring it to clone master only?

If I can avoid falling out to a manual clone in a script command I would like to. I want to drop the gh-pages content because I think that will have a much larger impact in the volume of data being brought down. The extra refs for the various branches don't add much weight I don't think.

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 1]
```diff
@@ -56,6 +56,9 @@ jobs:
           JavaVersion: '1.11'
 
     steps:
+      - checkout: self
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml]: Can this not be setup at a higher level to avoid duplication?

Comment (mitchdenny) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml]: There is a variable that can be set ```Agent.Source.Git.ShallowFetchDepth``` which does it globally. I'll do another PR which takes this route.

**Pull request merged**